src/npyffi/mod.rs

```diff
@@ -31,19 +36,61 @@ fn get_numpy_api<'py>(
     // so we can safely cache a pointer into its interior.
     forget(capsule);
 
+    API_VERSION.get_or_init(py, || unsafe {
+        #[allow(non_snake_case)]
+        let PyArray_GetNDArrayCFeatureVersion = api.offset(211) as *const extern "C" fn() -> c_uint;
+        (*PyArray_GetNDArrayCFeatureVersion)()
+    });
```

Comment:
This doesn't work if we happen to load the ufunc API first here (we get a nasty crash).

I would prefer to see this moved into a new public function:

```suggestion
fn is_numpy_2() -> bool {
    API_VERSION.get_or_init(py, || unsafe { PY_ARRAY_API.PyArray_GetNDArrayCVersion()}) >= API_VERSION_2_0
}
```

Reply:
Done. I changed places where we check the version to use this, and re-ordered the conditionals so that we always check `if is_numpy_2(py)`.